### ISSUE TITLE
Support Configuration via tox.ini

### ISSUE THREAD

**Issue by dciborow** [psf/black#2172]
**Is your feature request related to a problem? Please describe.** 
Currently, all of my test configurations are colocated within my tox.ini file for my entire project, except for my black configuration for line count = 120, which is stored in our pyproject.toml, and is the only setting configured there.

**Describe the solution you'd like** A clear and concise description of what you want to
happen.
Instead of having path a pyproject.yml and a tox.ini, I would like to be able to include the black configurations directly with tox.ini.

**Describe alternatives you've considered** A clear and concise description of any
alternative solutions or features you've considered.
I can't seem to find any single type of configuration file that works for pytest, pylint, bandit, flake8, and black. Tox is supported by the most of these, so I am trying to chase this down.

**Comment by dciborow** [comment]
Currently... we are able to put this in a plain tox.ini, and everything works.

![image](https://user-images.githubusercontent.com/9027725/116754506-d7969e80-a9d6-11eb-9a13-1a560c6ff7d3.png)


It is only when we add more things to tox.ini, that it stops working.
![image](https://user-images.githubusercontent.com/9027725/116754534-e41af700-a9d6-11eb-8680-f7345aabda63.png)

**Comment by JelleZijlstra** [comment]
I don't think we should do this. `pyproject.toml` is a standard(ish) way of configuring Python tools, while `tox.ini` is specific to Tox. Instead, other tools should move towards allowing configuration through `pyproject.toml`.

**Comment by dciborow** [comment]
@JelleZijlstra ,

Do you happen to know what about the pyproject.toml standard is different then the tox.ini standard? Particularly why the header names I use in tox.ini seem to break black's ability to process the tox.ini?


I would assume if someone added something like this to their pyproject.toml, it would also start to cause black to fail. But I am not familiar enough with pyproject.toml's standard to understand what makes it incompatible?

**Comment by JelleZijlstra** [comment]
Not too familiar with it, but obviously one is in TOML format (which is a standardized language) and one is in `.ini` format, which I think is just whatever Python's `configparser` module accepts.

**Comment by dciborow** [comment]
Also, there is negative traction towards expand pyproject.toml to other tools, particularly mypy. 

![image](https://user-images.githubusercontent.com/9027725/116755392-5a6c2900-a9d8-11eb-872d-f0ac830f56b5.png)

**Comment by JelleZijlstra** [comment]
Actually mypy is about to gain support for pyproject.toml (https://github.com/python/mypy/pull/10219).

**Comment by ichard26** [comment]
@dciborow I don't use Tox so I wouldn't know for sure, but the main difference causing them to be incompatible is probably that tox.ini uses the INI file format is 1) loosely defined AFAIK, and 2) is different enough from the [TOML file format](https://github.com/toml-lang/toml). pyproject.toml is just a normal TOML file who's contents has been somewhat standardized for Python's packaging usage (and now also tools under the `[tool.*]` namespace).  

I don't think our TOML parser can just ignore a misconfigured value / table. That sounds complicated and prone to errors. In your case, the problem is that booleans values must be lowercase in TOML. And allowing Black to run without the configuration sounds dangerous. It could format files it should never format because it was unable to read the exclude option configured in that invalid file.

**Comment by dciborow** [comment]
@ichard26 , 

Understand the points you are presenting. But to be super clear, the black configuration between the two files is identical.

So to be clear on the reverse point that I am trying to make... if a pyproject.toml is misconfigured outside of the [tools.black] section, you would expect black to fail? Even when the black section is completely correct and readable by black?
I think the below case fails just because [mypy] is missing "tool.". 
Where I am proposing black can ignore those sections, instead of failing on them.

pyproject.toml
```
[mypy]
python_version = 3.7
ignore_missing_imports = true
strict_optional = false
follow_imports = silent

[tool.black]
line-length = 120
fast = true
```

**Comment by JelleZijlstra** [comment]
We're just using a TOML parser. If the TOML is valid, we use it. If it's not valid, we don't, because we can't.

**Comment by dciborow** [comment]
okay... let me do some more debugging... see if its something else in my .ini thats messing this up.

It sounds like @JelleZijlstra , you would not expect the above sample to fail for black, given it was a valid TOML...?

**Comment by dciborow** [comment]
ahhh turns out this was my problem line.
tox.ini allows...
`follow_imports = silent`


But pyproject.yml wants
`follow_imports = 'silent'`

**Comment by felix-hilden** [comment]
For context, Tox is adding `pyproject.toml` configuration too (see tox-dev/tox#999). That doesn't mean that all the other tools using `tox.ini` will though, but that's something at least. I don't think expanding functionality back to `tox.ini` is worth it anymore.

**Comment by ambv** [comment]
We don't support multiple potential configuration files as this creates [terrible precedence rules](https://github.com/psf/black/issues/65#issuecomment-377587386). There's been [heated discussions](https://github.com/psf/black/issues/683) about similar ideas before, we're not doing it.

**Comment by mcarans** [comment]
This is a shame. I can configure flake8 and isort in tox.ini but not black.

**Comment by ikus060** [comment]
> This is a shame. I can configure flake8 and isort in tox.ini but not black.

Completely agree. Adding configuration to tox.ini is a defactor standard. Also `pyproject.toml` is fearly new and is not supported by older setuptools. In my project when trying to create the file `pyproject.toml` all our build start failing because of mysterious incompatibility.

**Comment by mcarans** [comment]
@ikus060 I've since moved to a full pyproject.toml setup by resurrecting flakehell, a wrapper around flake8 with nice pyproject.toml support that had become unmaintained, as [flakeheaven](https://github.com/flakeheaven/flakeheaven) in case this  is helpful to your project.
